bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -0,0 +1,247 @@
+        connectJob.ifPresent(j -> j.cancel(true));
+        connectJob = Optional.empty();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Shutting down Netatmo API bridge handler.");
+        servlet.ifPresent(servlet -> servlet.dispose());
+        servlet = Optional.empty();
+        connectApi.dispose();
+        freeConnectJob();
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Netatmo Bridge is read-only and does not handle commands");
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(NetatmoDiscoveryService.class);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T extends RestManager> @Nullable T getRestManager(Class<T> clazz) {
+        if (!managers.containsKey(clazz)) {
+            try {
+                Constructor<T> constructor = clazz.getConstructor(getClass());
+                T instance = constructor.newInstance(this);
+                Set<Scope> expected = instance.getRequiredScopes();
+                if (connectApi.matchesScopes(expected)) {
+                    managers.put(clazz, instance);
+                } else {
+                    logger.info("Unable to instantiate {}, expected scope {} is not active", clazz, expected);
+                }
+            } catch (SecurityException | ReflectiveOperationException e) {
+                logger.warn("Error invoking RestManager constructor for class {} : {}", clazz, e.getMessage());
+            }
+        }
+        return (T) managers.get(clazz);
+    }
+
+    public synchronized <T> T executeUri(URI uri, HttpMethod method, Class<T> clazz, @Nullable String payload,
+            int retryCount) throws NetatmoException {
+        try {
+            logger.trace("executeUri {}  {} ", method.toString(), uri);
+
+            Request request = httpClient.newRequest(uri).method(method).timeout(TIMEOUT_S, TimeUnit.SECONDS)
+                    .header(HttpHeader.CONTENT_TYPE, "application/x-www-form-urlencoded;charset=UTF-8");
+
+            String auth = connectApi.getAuthorization();
+            if (auth != null) {
+                request.header(HttpHeader.AUTHORIZATION, auth);
+            }
+
+            if (payload != null && (HttpMethod.POST.equals(method) || HttpMethod.PUT.equals(method))) {
+                InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+                try (InputStreamContentProvider inputStreamContentProvider = new InputStreamContentProvider(stream)) {
+                    request.content(inputStreamContentProvider, null);
+                }
+                if (!URL_API.contains(uri.getHost())) {
+                    request.getHeaders().remove(HttpHeader.CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, "application/json;charset=utf-8");
+                }
```

Comment:
> This looks a little tricky. Why not passing the content type as a method parameter ?
Yes, you're correct. This is used in a really small cases (when calling app.netatmo.com instead of api.netatmo.com, emulating the behaviour of the netatmo web / app client) for let see toggle the light of the presence, declare somebody inside... This would mean changing 95% of the call for very few cases. I already did a try to change this in a cleaner way but failed to test all cases.

Reply:
Just keep the method without parameter `contentType` znd set a default content type. And create another method with the parameter `contentType` and use it for only the 5% of your calls which require the non default content type.

Reply:
I see only 8 calls leading to a call to executeUri with POST method. None with a PUT method.
So it should not be too much difficult to make things clean.
I will submit a PR to you.

Reply:
PR submitted.

Reply:
Merged.